static/js/dynamic_text.js

```diff
@@ -0,0 +1,94 @@
+// This is a public instance of the `DynamicText` class.
+// The `DynamicText` object serves to resize text that is set through its
+// public methods so that the text won't ever overrun the parent container.
+var DynamicText = (function () {
+    // the initialization of the `DynamicText` instance.
+    var DynamicText = function ($parent) {
+        if (typeof jQuery !== "undefined" && $parent instanceof jQuery) {
+            this.parent = $parent[0];
+        } else if ($parent instanceof Node) {
+            this.parent = $parent;
+        }
+
+        this.node = document.createElement("span");
+        this.node.style.whiteSpace = "nowrap";
+
+        this.prev_content = this.parent.innerHTML;
+        this.parent.innerHTML = "";
+        this.parent.appendChild(this.node);
+
+        this.update();
+    };
+
+    // an object for private functions that are inaccessible to the outside
+    // world.
+    var internal_funcs = {
+        insertText: function (type, text, node, parent) {
+            if (type === DynamicText.prototype.TYPE.TEXT) {
+                node.innerText = text;
+            } else if (type === DynamicText.prototype.TYPE.HTML) {
+                node.innerHTML = text;
+            } else {
+                blueslip.warn("The method '" + type + "' is not a valid " +
+                              " DynamicText input method.");
```

Comment:
This should be `blueslip.error`, since it indicates a code bug that we should fix, not an unusual user input situation.  The general rule is that one should use `blueslip.warn` to flag things (but not send in an error report) and `blueslip.error` to send in an error report.